Use unique_ptr when class has clear ownership

troopa81 opened the pull request:
Mass refactor **on core only** using Clang LibTooling by introducing unique_ptr on "simple case" meaning when:
- Class has a raw pointer member
- We call delete on the member in destructor without any conditionnal statement
- QGIS still build without warning after the modification

It spots one issue
- QgsSettingsTreeNamedListNode copy constructor need to be deleted, it was default copying the raw pointer which could lead to double free. @3nids why do you to SIP_FORCE those ?

**Funded by [Security Project For QGIS](https://oslandia.com/en/security-project-for-qgis/)**

troopa81 committed:
refactor(core): Use unique_ptr when class has clear ownership

3nids commented:
> QgsSettingsTreeNamedListNode copy constructor need to be deleted, it was default copying the raw pointer which could lead to double free. @3nids why do you to SIP_FORCE those ?

I think it was required to build the bindings. But if it works without it, I don't have any objection.

troopa81 committed:
fix(core): fix windows build

troopa81 commented:
> I'm curious why QgsDockableWidgetHelper from https://github.com/qgis/QGIS/pull/61483 didn't classify here?

I forgot to mention that I refactored only core (I updated the PR comment), I would do it for other modules then.
